internal/hash/checksum.go

```diff
@@ -325,12 +342,15 @@ func (c *Checksum) AppendTo(b []byte, parts []byte) []byte {
 	b = append(b, crc...)
 	if c.Type.Is(ChecksumMultipart) {
 		var checksums int
+		if c.WantParts > 0 && !c.Type.Is(ChecksumIncludesMultipart) {
```

Comment:
it seems from the code that c.WantParts is always zero when AppendTo is called from erasure-multipart.go, not sure if I am missing something

Reply:
@vadmeste You are right. `WantParts` is only used transiently now (comes from request, validated against actual) and not stored. 

I just didn't want this to be "hanging" in case that was changed some day, so there is a way for it to be serialized in case we at some point need to send it in and RPC call or similar.